osbrain/tests/test_agent.py

```diff
@@ -545,6 +545,36 @@ def test_close(nsproxy):
     assert not a0.has_socket('pub')
 
 
+def test_close_socket_poller_cleanup(nsproxy):
+    """
+    Closed sockets should be cleaned-up from the poller.
+
+    When Bob connects to Alice, a new socket is registered in Bob's poller.
+    Once we close that socket, it should no longer be registered in the poller.
+    """
+    def rep_handler(agent, message):
+        return 'OK'
```

Comment:
You can merge the other PR and use the new echo handler to fix the coverage diff issue here.

Reply:
Yeah, thanks for updating it! :blush:

Already rebased against master and added a fixup to change that.